File: Leetcode/1456 Maximum Number of Vowels in a Substring of Given Length.py
VOWELS = 'aeiou'

class Solution:
    def maxVowels(self, s, k):
        vowels, max_vowels = 0, 0
        for i, c in enumerate(s):
            if c in VOWELS:
                vowels += 1
            if i >= k and s[i - k] in VOWELS:
                vowels -= 1
            max_vowels = max(max_vowels, vowels)
        return max_vowels

File: Leetcode/1456 Maximum Number of Vowels in a Substring of Given Length.cs
public class Solution
{
    private static int NVowel(char c) => "aeiou".Contains(c) ? 1 : 0;

    public int MaxVowels(string s, int k)
    {
        int vowelCount = s.Take(k).Select(NVowel).Sum();
        int maxVowelCount = vowelCount;
        for (int i = k; i < s.Length; i++)
        {
            vowelCount += NVowel(s[i]) - NVowel(s[i - k]);
            maxVowelCount = Math.Max(maxVowelCount, vowelCount);
        }
        return maxVowelCount;
    }
}
